docs/components/file-upload.md

```diff
@@ -0,0 +1,213 @@
+# File Upload
+
+**KFileUpload** provides a wrapper around general `input` with `type=field` for `file upload` or `image upload`.
+ 
+<KCard>
+  <template v-slot:body>
+    <KFileUpload label="Upload File" :label-attributes="{ help: 'I use the KLabel `help` prop' }" help="Additional text." icon="close" :accept="['.yml', '.yaml', '.json', '.md', '.markdown', 'image/gif', 'image/jpg', 'image/jpeg', 'image/ico', 'image/png']" appearance="file" removable />
+  </template>
+</KCard>
+
+```html
+<KFileUpload label="Upload File" :label-attributes="{ help: 'I use the KLabel `help` prop' }" help="Additional text." icon="close" :accept="['.yml', '.yaml', '.json', '.md', '.markdown', 'image/gif', 'image/jpg', 'image/jpeg', 'image/ico', 'image/png']" appearance="file" removable />
+```
+
+## Props
+
+### appearance
+
+KFileUpload has default appearance of type `file`, and can be switched either to - `file`, `image`.
+
+<KCard>
+  <template v-slot:body>
+    <KFileUpload label="Upload File" appearance="file" :label-attributes="{ help: 'I use the KLabel `help` prop' }" help="Additional files can be uploaded from HomePage." icon="close" :accept="['.yml', '.yaml', '.json', '.md', '.markdown', 'image/gif', 'image/jpg', 'image/jpeg', 'image/ico', 'image/png']" removable @file-added="(i) => { fileName = i[0].name, fileSize = i[0].size }" @file-removed="() => { fileName = '', fileSize = '' }" />
+  </template>
+</KCard>
+
+<div class="mt-6">Emitted value: <pre v-if="fileName && fileSize" class="emitted-value">{{ `File Name: ${fileName}` }} and {{ `File Size: ${fileSize}` }}</pre></div>
+
+```html
+<KFileUpload label="Upload File" appearance="file" :label-attributes="{ help: 'I use the KLabel `help` prop' }" help="Additional files can be uploaded from HomePage." icon="close" :accept="['.yml', '.yaml', '.json', '.md', '.markdown', 'image/gif', 'image/jpg', 'image/jpeg', 'image/ico', 'image/png']" removable />
+```
+
+<KCard class="mt-6">
+  <template v-slot:body>
+    <KFileUpload appearance="image" icon="image" :accept="['.yml', '.yaml', '.json', '.md', '.markdown', 'image/gif', 'image/jpg', 'image/jpeg', 'image/ico', 'image/png']" placeholder="Upload new image (Max 4 MB)" @file-added="(i) => { imageName = i[0].name, imageSize = i[0].size }" >
+    </KFileUpload>
+  </template>
+</KCard>
+
+<div class="mt-6">Emitted value: <pre v-if="imageName && imageSize" class="emitted-value">{{ `File Name: ${imageName}` }} and {{ `File Size: ${imageSize}` }}</pre></div>
+
+```html
+<KFileUpload appearance="image" icon="image" :accept="['.yml', '.yaml', '.json', '.md', '.markdown', 'image/gif', 'image/jpg', 'image/jpeg', 'image/ico', 'image/png']" placeholder="Upload new image (Max 4 MB)" />
+```
+
+### label
+
+String to be used as the `KFileUpload` label when `appearance` is of type `file`.
```

Comment:
I haven't looked at the code but I wouldn't restrict this prop to only be allowed with `file` appearance. In the dev portal use case they can choose to not use the prop, but we should allow use with `appearance` `image`